### ISSUE TITLE
consideration for use of single key shortcuts

### ISSUE THREAD

**Issue by DavidMacDonald** [w3c/aria-practices#450]
There is a section on adding keyboard shortcuts.
https://www.w3.org/TR/wai-aria-practices/#kbd_shortcuts_assigning

It talks about single key shrotcuts.

It should be revised in anticipation of the new WCAg 2.1 SC on keyboard shortcuts. See here.
https://rawgit.com/w3c/wcag21/target-size/guidelines/sc/21/target-size.html

**Explanation here**

https://rawgit.com/w3c/wcag21/character-key-shortcuts/understanding/21/character-key-shortcuts.html

**Comment by jnurthen** [comment]
@DavidMacDonald can you tell me exactly where it talks about single key shortcuts. I couldn't find them mentioned.

**Comment by DavidMacDonald** [comment]
@jnurthen 
The following guidance is recommending a single key shortcut "s" for save. 

> When choosing the keys to assign to a shortcut, there are many factors to consider.
> * Making the shortcut easy to learn and remember by using a mnemonic (e.g., "S" for "Save") or following a logical or spacial pattern.

https://www.w3.org/TR/wai-aria-practices/#kbd_shortcuts_assigning

WCAG 2.1 will probably have the SC ["Character Key shortcuts"](https://rawgit.com/w3c/wcag21/character-key-shortcuts/guidelines/sc/21/character-key-shortcuts.html) That advises against this.

**Comment by jnurthen** [comment]
I don't agree that this is recommending a single key shortcut. It is saying to choose the letter for a shortcut to be easy to remember.
However if people reading it are interpreting it that way then we have an issue which needs to be resolved. Potentially a fix will be to add a section after "Browser Key Conflicts" dealing with "Single Key Conflicts" where we talk about a requirement to turn single key shortcuts.

**Comment by DavidMacDonald** [comment]
@jnurthen 
I don't know any other way I could read it. It says use "s" as the keyboard shortcut for "save"?

This is a single key shortcut, no? I think it would require a rethink of the bullet to be split up something like this:

> * Following a logical or spacial pattern
> * If there is a setting to turn off single key shortcuts for voice users, then making the shortcut easy to learn and remember by using a mnemonic (e.g., "S" for "Save").

Then the phase single key shortcut would have a link to the WCAG SC or a definition or explanation etc...

**Comment by mcking65** [comment]
@DavidMacDonald, thanks for the specific reference. I intended that to be ctrl-s. Nevertheless, it would be good to address single-key shortcut considerations somewhere in there.

**Comment by mcking65** [comment]
@DavidMacDonald , With commit 193bd08, I removed the text that triggered your concern.

However, that does not address the larger issue of guidance related to single-key shortcuts. I am punting that down the road to release 2 of the practices by assigning this issue to that milestone.
